src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -0,0 +1,206 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Sarif.Visitors;
+using Microsoft.TeamFoundation.WorkItemTracking.WebApi.Models;
+using Microsoft.VisualStudio.Services.CircuitBreaker;
+using Microsoft.WorkItems;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.WorkItems
+{
+    public class SarifWorkItemFiler
+    {
+        public SarifWorkItemFiler() { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SarifWorkItemFiler"> class.</see>
+        /// </summary>
+        /// <param name="filingClient">
+        /// A client for communicating with a work item filing host (for example, GitHub or Azure DevOps).
+        /// </param>
+        /// <param name="filingContext">
+        /// A root context object that configures the work item filing operation.
+        /// </param>
+        public SarifWorkItemFiler(SarifWorkItemContext filingContext)
+        {
+            this.FilingContext = filingContext ?? throw new ArgumentNullException(nameof(filingContext));
+            this.FilingClient = FilingClientFactory.CreateFilingTarget(filingContext.ProjectUri);
+        }
+
+        public SarifWorkItemContext FilingContext { get; set; }
+
+        public FilingClient FilingClient { get; set; }
+
+        public virtual void FileWorkItems(Uri sarifLogFileLocation)
+        {
+            sarifLogFileLocation = sarifLogFileLocation ?? throw new ArgumentNullException(nameof(sarifLogFileLocation));
+
+            if (sarifLogFileLocation.IsAbsoluteUri && sarifLogFileLocation.Scheme == "file:")
+            {
+                FileWorkItems(File.ReadAllText(sarifLogFileLocation.LocalPath));
+            }
+
+            throw new NotImplementedException();
```

Comment:
>) [](start = 46, length = 1)

This needs a message, otherwise the victim things that `FileWorkItems` isn't implemented at all. ("FileWorkItems only supports file: URIs")